### PR TITLE
Implement create ticket feature

### LINKED ISSUE #396: Add button in incident view to open a new issue in ticket system
https://github.com/Uninett/Argus/issues/474 introduces an endpoint to get an url to a created issue in an issue system. 

This endpoint should be used to add a button to the incident view to open this link in a new tab and show any errors that might happen.

### PR DESCRIPTION
**Pull request by podliashanyk**
Closes #396, #449

### CODE REVIEW

**Comment by podliashanyk** [comment]
Remaining tasks:

- [ ] manual test against an actual ticket plugin
- [ ] tests
- [ ] update changeling

**Comment by podliashanyk** [comment]
Changes requested after IRL conversation with @hmpf:

- [ ] If incident has no ticket, click on _Create Ticket_ button should:
  - [ ] fire a generate ticket event if user has not pasted any url into the input field
  - [ ] save the ticket url that user has pasted into the input field
- [ ] If incident has a ticket:
  - [ ] text in the _Ticket_ input field should be a clickable url
  - [ ] _Create Ticket_ button is changed to the _Edit ticket_ button. Click on the _Edit ticket_ button prompts a dialog where user can manually change a ticket url, or paste a new one

**Comment by podliashanyk** [comment]
Currently implemented functionality:

- [ ] If incident has no ticket, click on _Create Ticket_ button:
  - [ ] fires a generate ticket event if user has not pasted any url into the input field (prompts a confirm dialog)
  - [ ] save the ticket url that user has pasted into the input field
- [ ] If incident has a ticket:
  - [ ] text in the _Ticket_ input field is a clickable url
  - [ ] _Create Ticket_ button is changed to the _Edit ticket_ button. Click on the _Edit ticket_ button make ticket field to an editable input field, and changes button text to _Save ticket_


Remaining tasks:

- [ ] Fix CSS
- [ ] Fix failing tests
- [ ] Write more tests
- [ ] Update changelog

File: src/state/hooks.tsx [hunk 1]
```diff
@@ -99,4 +107,37 @@ export const useTimeframe = (): [TimeframeStateType, UseTimeframeActionType] =>
...
+    ticketState,
+    {
+      initTicketState: initTicketStateCallback,
+      changeUrl: changeUrlCallback,
+      invalidUrl: invalidUrlCallback,
+      manuallyEditTicket: manuallyEditTicketCallback,
+      resetTicketState: resetTicketStateCallback,
+    },
+  ];
 };
```
Comment (hmpf) [inline src/state/hooks.tsx:143]: Meep-meep, maybe we should have pre-commit add missing newlines?

File: src/state/reducers/ticketurl.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        default:
+            return {...state};
+    }
+};
+
+export const initTicketState = makeAction<TicketType.InitTicketState, string | undefined>(TicketType.InitTicketState);
+export const changeUrl = makeAction<TicketType.ChangeUrl, string | undefined>(TicketType.ChangeUrl);
+export const invalidUrl = makeActionWithoutPayload<TicketType.InvalidUrl>(TicketType.InvalidUrl);
+export const manuallyEditTicket = makeActionWithoutPayload<TicketType.ManuallyEditTicket>(TicketType.ManuallyEditTicket);
+export const resetTicketState = makeAction<TicketType.ResetTicketState, string | undefined | null>(TicketType.ResetTicketState);
```
Comment (hmpf) [inline src/state/reducers/ticketurl.ts:74]: Another missing newline.

**Review by hmpf — CHANGES_REQUESTED** [review]
This new feature is not mentioned in MANUAL.md. We could punt that to another PR, but then we should have an issue about it so we don't forget it.

**Review by hmpf — APPROVED** [review]

**Pull request merged**